gp-includes/things/translation.php

```diff
@@ -415,15 +415,54 @@ public function for_translation( $project, $translation_set, $page, $filters = a
 
 		$sql_sort = sprintf( $sort_by, $sort_how );
 
-		$limit = $this->sql_limit_for_paging( $page, $this->per_page );
+		/**
+		 * Filter the translation sets to allow falling back to another translation set.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param null Default for no additional translation set.
+		 * @param GP_Translation_Set $translation_set Current translation set.
+		 */
+		$fallback_translation_set = apply_filters( 'gp_for_translation_fallback_translation_set', null, $translation_set );
+
+		$translation_sets = array( $translation_set->id );
+
+		if ( $fallback_translation_set ) {
+			$limit = $this->sql_limit_for_paging_with_fallback( $page, $this->per_page );
+			$translation_sets[] = (int) $fallback_translation_set;
+		} else {
+			$limit = $this->sql_limit_for_paging( $page, $this->per_page );
+		}
+
+		$translation_sets = implode( ',', $translation_sets );
 
 		$sql_for_translations = "
 			SELECT SQL_CALC_FOUND_ROWS t.*, o.*, t.id as id, o.id as original_id, t.status as translation_status, o.status as original_status, t.date_added as translation_added, o.date_added as original_added
 			FROM $wpdb->gp_originals as o
-			$join_type JOIN $wpdb->gp_translations AS t ON o.id = t.original_id AND t.translation_set_id = " . (int) $translation_set->id . " $join_where
+			$join_type JOIN $wpdb->gp_translations AS t ON o.id = t.original_id AND t.translation_set_id IN( " . $wpdb->esc_like( $translation_sets ) . " ) $join_where
```

Comment:
Thanks, that's one ugly solution ;-) but I guess working around the paging issue might be worst. I'll look into it.